File: 125. Valid Palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        s = s.lower()
        start = 0
        end = len(s) - 1
        while start<=end:
            if s[start].isalnum() and s[end].isalnum():
                if s[start] != s[end]:
                    return False
                start += 1
                end -= 1
            else:
                if not s[start].isalnum():
                    start += 1
                elif not s[end].isalnum():
                    end -= 1
        return True

File: 125. Valid Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        for (int i = 0; i < s.length(); i++)
            s[i] = tolower(s[i]);
        int start = 0;
        int end = s.length() - 1;
        while (start <= end)
        {
            if (isalnum(s[start]) && isalnum(s[end]))
            {
                if (s[start] != s[end])
                    return false;
                start ++;
                end --;
            }
            else
            {
                if (!isalnum(s[start]))
                    start ++;
                else
                    end --;
            }
        }
        return true;
    }
};
